plugins/chat/spec/services/auto_remove/handle_destroyed_group_spec.rb

```diff
@@ -0,0 +1,255 @@
+# frozen_string_literal: true
+
+RSpec.describe Chat::Service::AutoRemove::HandleDestroyedGroup do
+  describe ".call" do
+    subject(:result) { described_class.call(params) }
+
+    let(:params) { { destroyed_group_user_ids: [admin_1.id, admin_2.id, user_1.id, user_2.id] } }
+    fab!(:user_1) { Fabricate(:user) }
+    fab!(:user_2) { Fabricate(:user) }
+    fab!(:admin_1) { Fabricate(:admin) }
+    fab!(:admin_2) { Fabricate(:admin) }
+
+    fab!(:dm_channel_1) { Fabricate(:direct_message_channel, users: [admin_1, user_1]) }
+    fab!(:dm_channel_2) { Fabricate(:direct_message_channel, users: [user_1, user_2]) }
+
+    fab!(:channel_1) { Fabricate(:chat_channel) }
+    fab!(:channel_2) { Fabricate(:chat_channel) }
+
+    context "when chat is not enabled" do
+      let(:new_allowed_groups) { "1|2" }
+
+      before { SiteSetting.chat_enabled = false }
+
+      it { is_expected.to fail_a_policy(:chat_enabled) }
+    end
+
+    context "when chat is enabled" do
+      before { SiteSetting.chat_enabled = true }
+
+      context "if none of the group_user_ids users exist" do
+        before { User.where(id: params[:destroyed_group_user_ids]).destroy_all }
+
+        it "fails to find scoped_users model" do
+          expect(result).to fail_to_find_a_model(:scoped_users)
+        end
+      end
+
+      describe "step remove_users_outside_allowed_groups" do
+        context "when chat_allowed_groups is empty" do
+          let(:action) { UserHistory.where(custom_type: "chat_auto_remove_membership").last }
+          before { SiteSetting.chat_allowed_groups = "" }
```

Comment:
We have two `before` blocks defined here. We should move this line into the `before` block line 43.